src/main/app/case/CosApiClient.ts

```diff
@@ -339,6 +339,61 @@ export class CosApiClient {
     });
     return response.data;
   }
+
+  public async retrieveCaseHearingsByCaseId(caseId: string, user: UserDetails): Promise<CaseWithId> {
+    if (!caseId || !user) {
+      return Promise.reject(new Error('retrieveCaseHearingsByCaseId - Case id must be set and user must be set'));
+    }
+    // const response = await Axios.get(config.get('services.cos.url') + `/${caseId}`, {
+    //   headers: {
+    //     Authorization: 'Bearer ' + user.accessToken,
+    //     serviceAuthorization: getServiceAuthToken(),
+    //     Accept: 'application/json',
+    //     'Content-Type': 'application/json',
+    //   },
+    // });
+
+    //req.session.userCase = {id:'1234', state: State.Holding};
+    //req.session.userCase.hearingCollection = [];
+
+    const hearingCollectionMockedData = [
+      {
+        prev: [
+          {
+            date: 'Monday 28 June 2021',
+            time: '9am',
+            typeOfHearing: 'In person',
+            courtName: 'Bristol family court',
+            courtAddress: 'Bristol Avenue, Bristol, BL1 2A3',
+            hearingOutcome: 'HearingOutcome.pdf',
+          },
+          {
+            date: 'Friday 25 June 2021',
+            time: '11am',
+            typeOfHearing: 'Video',
+            courtName: 'Bristol family court',
+            courtAddress: 'Bristol Avenue, Bristol, BL1 2A3',
+            hearingOutcome: 'HearingOutcome.pdf',
+          },
+        ],
+        next: {
+          date: 'Wednesday 29 June 2021',
+          time: '10am',
+        },
+      },
+    ];
+
+    // return {
+    //   id: response.data.id,
+    //   state: response.data.state,
+    //   ...fromApiFormat(response.data),
+    // };
+    return {
+      id: '1234567890',
+      state: State.AwaitingHWFDecision,
+      hearingCollection: hearingCollectionMockedData,
+    };
```

Comment:
how will it work once we get the real data? awaitngHWFDecision? is that even a real state?

Reply:
code is updated, userCase is passed as input parameter to the method which contains the details of case id, case state. Once the backend api is ready we can integrate accordingly.